extracted/block_13.py

```diff
@@ -0,0 +1,53 @@
+# AAtypes
+# Args : string of AA
+# Compute the number of aminoacids that are POLAR, SMALL, HYDROPHOBIC in that order
+
+# Example: AAtypes("GALMFWKQESCYHRNDTPVI")
+# Other characters than above will produce a Bad Sequence Exception
+# (Case-insensitive)
+
+# Define function(arg)
+def AAtypes(aaseq):
+    
+    # Ensure Upper, this is more for downstream analysis
+    aaseq = aaseq.upper()
+    
+    # Ensure characters are aminoacids
+    for i in aaseq:
+        if i not in "GALMFWKQESCYHRNDTPVI":
+            raise BadSequenceException("Input string must be amino acid one character codes only, no special characters")
+    
+    # Define which AA's are polar, small, hydrophobic
+    polar = "YWHKREDTSCNQ"
+    small = "VCGSTAP"
+    hydrophobic = "FYWHKMCTAILV"
+    
+    # Counts for the properties
+    polar_aa = 0
+    small_aa = 0
+    hydrophobic_aa = 0
+    
+    # Iterate through, adding to the counts
+    for i in aaseq:
+        if i in polar:
+            polar_aa += 1
+        if i in small:
+            small_aa += 1
+        if i in hydrophobic:
```

Comment:
if these are mutually exclusive, consider using an `elif` to save the extra evaluations